memif-attach/main.go

```diff
@@ -18,59 +18,102 @@ package main
 
 import (
 	"context"
+	"net"
 	"time"
-
+	// "io"
+	"encoding/binary"
+	"bufio"
+	"bytes"
 	"git.fd.io/govpp.git/api"
 	"github.com/edwarnicke/govpp/binapi/memif"
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/edwarnicke/govpp/binapi/interface_types"
 	"github.com/edwarnicke/log"
 	"github.com/edwarnicke/vpphelper"
 )
 
+// HelloReplyMsg type
+type HelloMsg struct {
+	max_region int16 
+	max_m2s_ring int16 
+	max_s2m_ring int16
+	min_version int16 
+	max_version int16
+	max_log2_ring_size int8 
+}
+// UnmarshalBinary Function
+func (replyMsg *HelloMsg) UnmarshalBinary(data []byte) error {
+	buf := bytes.NewReader(data)
+	err := binary.Read(buf, binary.BigEndian, replyMsg)
+	return err
+}
 func main() {
 	ctx, cancel := context.WithCancel(context.Background())
-
+	
 	// Connect to VPP with a 1 second timeout
 	connectCtx, _ := context.WithTimeout(ctx, time.Second)
 	conn, vppErrCh := vpphelper.StartAndDialContext(connectCtx)
 	exitOnErrCh(ctx, cancel, vppErrCh)
-
+	
 	// Create a RPC client for the memif api
 	isClient := true
 	// Add a memif socket
-	socketID, _ := createMemifSocket(ctx, conn)
+	socketID, socketAddr, _:= createMemifSocket(ctx, conn)
 	// Create a memif interface
 	mode := memif.MEMIF_MODE_API_IP
 	createMemif(ctx, conn, socketID, mode, isClient)
-	// Delete a memif socket
+	
+	// Dump memif info
 	dumpMemif(ctx, conn)
+	
+	// connect to VPP
+	conn_client, err := net.Dial("unixpacket", socketAddr)
+	if err != nil {
+		log.Entry(ctx).Fatalln("ERROR: connect to VPP master failed:", err)
+	}
+	// read hello message from master vpp
+	helloMsg := &HelloMsg{}
+	reader := bufio.NewReader(conn_client)
+	buf, _, readErr := reader.ReadLine()
+	if readErr != nil {
+		log.Entry(ctx).Fatalln("Read Error", readErr)
+	}
+
+	helloErr := helloMsg.UnmarshalBinary(buf) // decode reply message
```

Comment:
You might want to try:
```suggestion
helloErr := binary.Read(reader, binary.BigEndian, replyMsg)
```